File: 231110/겹치지 않는 사각형의 넓이/area-of-non-overlapping-rectangle.py
rec=[[0 for j in range(2001)] for i in range(2001)]

for _ in range(2):
    x1,y1,x2,y2=map(int,input().split())
    for i in range(x1,x2):
        for j in range(y1,y2):
            rec[i][j]=1

x1,y1,x2,y2=map(int,input().split())

for i in range(x1,x2):
    for j in range(y1,y2):
        rec[i][j]=2

ans=0
for i in rec:
    for j in i:
        if j==1:
            ans+=1

print(ans)

File: 251204/겹치지 않는 사각형의 넓이/area-of-non-overlapping-rectangle.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const rects = input.map((line)=>line.split(' ').map(Number));

const OFFSET = 1000;
const MAX_R = 2000;

const arr = Array.from({length:MAX_R+1}, ()=> Array(MAX_R+1).fill(0));


for(let i=0; i<3; i++){
    let [x1, y1, x2, y2] = rects[i];
    x1+=OFFSET; y1+=OFFSET;
    x2+=OFFSET; y2+=OFFSET;
    for(let j=x1; j<x2; j++){
        for(let k=y1; k<y2; k++){
            if(i===2){
                arr[j][k]=0;
            }else{
                arr[j][k]=1;
            }
        }
    }
}

let cnt=0;
for(let row of arr){
    cnt+=row.filter((e)=>e===1).length;
}

console.log(cnt)